packages/core/src/modules/credentials/CredentialsModule.ts

```diff
@@ -354,6 +355,14 @@ export class CredentialsModule<
     const service = this.getService(credentialRecord.protocolVersion)
     await service.updateState(credentialRecord, CredentialState.Declined)
 
+    //  if credential was offered by established connection, attempt to send problem report upon credential decline
+    if (credentialRecord.connectionId) {
```

Comment:
We can also respond to connectionless exchange using the same technique we do as with the other methods (service decorator). Maybe we can add that here also?

It should at least throw an error I think instead of silently not do anything without a connection id.